### PR TITLE
Fixes #17002 - Update facts in candlepin

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
Note that i intentionally updated these outside of dynflow in order to limit the amount of input data in the dynflow tasks

File: app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb [hunk 1]
```diff
@@ -253,6 +253,7 @@ def facts
         sync_task(::Actions::Katello::Host::Update, @host, rhsm_params.except(:facts))
         Katello::Host::SubscriptionFacet.update_facts(@host, rhsm_params[:facts]) unless rhsm_params[:facts].nil?
         update_host_registered_through(@host, request.headers)
+        @host.subscription_facet.candlepin_consumer.update(rhsm_params.slice(:facts))
```
Comment (ehelms) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb]: Should we also avoid updating the facts here if they are nil similar to line 254?

**Comment by thomasmckay** [comment]
This handles the /rhsm route but the subscription_facet_attributes in PUT /api/hosts does not update the facts. Is that something to be done in this PR or is that a different issue?

File: app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb (hunk 1 above)
Comment (thomasmckay) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb]: +1

**Comment by jlsherrill** [comment]
@thomasmckay updated to push everything back into dynflow, but added a finalize to 'trim' the facts.  

I investigated using middleware to try to make this filtering more generic but that was just adding complexity without much benefit, so here i'm just trimming it in finalize()

**Comment by thomasmckay** [comment]
ack, thanks!

**Pull request merged**